libdnf/base/transaction.cpp

```diff
@@ -326,10 +271,15 @@ TransactionPackage Transaction::Impl::make_transaction_package(
     std::map<Id, std::vector<Id>> & replaced) {
     auto obs = solved_goal.list_obsoleted_by_package(id);
 
-    libdnf_assert(!obs.empty(), "No obsoletes for {}", transaction_item_action_to_string(action));
-
     rpm::Package new_package(base, rpm::PackageId(id));
-    auto reason = new_package.get_reason();
+
+    transaction::TransactionItemReason reason;
+    if (action == TransactionPackage::Action::INSTALL) {
+        reason = solved_goal.get_reason(id);
```

Comment:
I've solved the problem with a bit different query (well, similarly to the inner query here, but filtering for NA instead of just name. Not sure how much of a performance issue this could be, but OTOH it's independent on whether a package is installonly. Though, maybe we should have a `Package::is_installonly()` method which would be helpful here?